core/src/main/java/org/infinispan/commands/read/EntrySetCommand.java

```diff
@@ -57,6 +57,10 @@ public LoadType loadType() {
 
    @Override
    public Set<CacheEntry<K, V>> perform(InvocationContext ctx) throws Throwable {
+      Object lockOwner = ctx.getLockOwner();
+      if (ctx.getLockOwner() != null) {
+         return new BackingEntrySet<>(cache.getAdvancedCache().lockAs(lockOwner));
```

Comment:
Sounds like this could work. Just do what you think is the best and provide a test :)